pyprep/utils.py

```diff
@@ -73,28 +75,44 @@ def _mat_quantile(arr, q, axis=None):
     This function mimics MATLAB's logic to produce identical results.
 
     """
-
-    def _mat_quantile_1d(arr, q):
-        arr = arr[~np.isnan(arr)]
-        n = len(arr)
-        if n == 0:
-            return np.NaN
-        elif n == 1:
-            return arr[0]
-        else:
-            q_adj = ((q - 0.5) * n / (n - 1)) + 0.5
-            return np.quantile(arr, np.clip(q_adj, 0, 1))
-
-    # Make sure inputs are both Numpy arrays
-    arr = np.asarray(arr)
-    q = np.asarray(q, dtype=np.float64)
-
-    if axis is not None and len(arr.shape) > 1:
-        # If an axis is specified, calculate quantiles along it
-        return np.apply_along_axis(_mat_quantile_1d, axis, arr, q)
+    # Sort the array in ascending order along the given axis (any NaNs go to the end)
+    # Return NaN if array is empty.
+    arr_sorted = np.sort(arr, axis=axis)
+    if arr_sorted.size == 0:
+        return np.NaN
```

Comment:
You could check this before sorting, right? no need to sort if you are going to return NaN anyhow.

Reply:
I was thinking that I wanted it to be coerced to `np.ndarray` format by `sort` already before checking size, but I realize now that `len(arr)` would work just as well for empty lists and arrays. Will fix in the next commit!